modules/azure/resourceid.go

```diff
@@ -0,0 +1,36 @@
+package azure
+
+import (
+	"errors"
+	"strings"
+)
+
+// GetNameFromResourceID gets the Name from an Azure Resource ID
+func GetNameFromResourceID(resourceID string) string {
+	lastValue, err := GetSliceLastValueE(resourceID, "/")
+	if err != nil {
+		return ""
+	}
+	return lastValue
+}
+
+// GetSliceLastValueE will take a source string and returns the last value when split by the seperaror char
+func GetSliceLastValueE(source string, seperator string) (string, error) {
+	if !(len(source) == 0 || len(seperator) == 0 || !strings.Contains(source, seperator)) {
+		tmp := strings.Split(source, seperator)
+		return tmp[len(tmp)-1], nil
+	}
+	return "", errors.New("invalid input or no slice available")
```

Comment:
NIT: reading "not" logic is always harder mentally. Consider flipping the structure:

```go
if (len(source) == 0 || ...) {
  return "", errors.New()
}

tmp := strings...
```

Reply:
Also, we declare explicit types for each error so that in tests, you can check for those exact types. Could you replace `errors.New` with a custom type?

Reply:
Removed not logic and created custom error, NewSliceValueNotFoundError